api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirImmunizationServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.impl;
+
+import static org.hl7.fhir.r4.model.Patient.SP_IDENTIFIER;
+import static org.openmrs.module.fhir2.api.translators.impl.ImmunizationTranslatorImpl.immunizationGroupingConcept;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.hl7.fhir.r4.model.Immunization;
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.PatientService;
+import org.openmrs.module.fhir2.api.FhirImmunizationService;
+import org.openmrs.module.fhir2.api.translators.ImmunizationTranslator;
+import org.openmrs.module.fhir2.api.translators.impl.ImmunizationObsGroupHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FhirImmunizationServiceImpl implements FhirImmunizationService {
+	
+	@Autowired
+	private ImmunizationTranslator translator;
+	
+	@Autowired
+	private ObsService obsService;
+	
+	@Autowired
+	private EncounterService encounterService;
+	
+	private PatientService patientService;
+	
+	private ImmunizationObsGroupHelper helper;
+	
+	@Autowired
+	public void setPatientService(PatientService patientService) {
+		this.patientService = patientService;
+	}
+	
+	@Autowired
+	public void setImmunizationHelper(ConceptService conceptService) {
+		this.helper = new ImmunizationObsGroupHelper(conceptService);
+	}
+	
+	@Override
+	public Immunization get(String uuid) {
```

Comment:
Yes, I'm about to push those changes. Actually I had to revert to using `ReferenceAndListParam` rather than just `ReferenceParam` in order to being able to leverage `FhirObservationDao` out of the box:
```java
public Collection<Immunization> searchImmunizations(ReferenceAndListParam patientParam, SortSpec sort) { ...
```